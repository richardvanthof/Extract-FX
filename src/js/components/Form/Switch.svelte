<script lang="ts">
    import { globals } from "../../global-vars/globals.svelte";
    import {handleClick} from "@/js/helpers/helpers";
    type Props = {
        value: string,
        options: string[],
        callback: (selection: 'file'|'sequence') => void
    }
    let {options, value, callback}: Props = $props();

</script>


<ul class="switch-input">
    {#each options as option}
    <button 
    class="switch-option {value.toLowerCase() === option.toLowerCase() ? 'active' : ''}"
    onclick={(e:Event) => handleClick(e, callback(option.toLocaleLowerCase()))} >
        {option}
    </button>
    {/each}
</ul>


<style lang="scss">
    ul {
        margin: 0;
        margin-bottom: 0px;
        padding: 0;
        display: flex;
        padding: 0.4em;
        background: var(--bg-dark);
        justify-content: space-evenly;
        margin-bottom: 1em;
        column-gap: .5em;
        border-radius: var(--border-radius-outer);
    }

    button {
        padding: .8em 1.5em;
        background: var(--bg-light);
        list-style: none;
        width: 100%;
        text-align: center;
        justify-content: space-evenly;
        border: var(--border-styling);
        color: var(--text);
        cursor: pointer;
        border-radius: var(--border-radius-inner);
        &:active {
            background: var(--bg-btn-active);
        }
    }

    .active {
        background: var(--bg-selected);
        border: var(--border-styling-active);
        &:active {
            background: var(--bg-btn-active);
        }
    }
</style>