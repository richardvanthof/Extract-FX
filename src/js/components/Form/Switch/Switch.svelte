<script lang="ts">
    import {handleClick} from '@/js/helpers/helpers.svelte';
    type Props = {
        value: string,
        options: string[],
        callback: (selection: string) => void,
        label?: string,
    }
    let {options, value, callback, label}: Props = $props();
    const id = label && 'switch-' + label.replace(' ', '-').toLowerCase();

</script>

{#if label}
<label for={id}>{label}</label>
{/if}
<ul class="switch-input" {id}>
    {#each options as option}
    <button 
    class="switch-option {value.toLowerCase() === option.toLowerCase() ? 'active' : ''}"
    onclick={(e:Event) => handleClick(e, callback(option.toLocaleLowerCase()))} >
        {option}
    </button>
    {/each}
</ul>


<style lang="scss">
    ul {
        margin: 0;
        margin-bottom: 0px;
        padding: 0;
        display: flex;
        padding: 0.4em;
        background: var(--bg-dark);
        justify-content: space-evenly;
        margin-bottom: 1em;
        column-gap: .5em;
        border-radius: var(--border-radius-outer);
    }

    button {
        padding: .8em 1.5em;
        background: var(--bg-light);
        list-style: none;
        width: 100%;
        text-align: center;
        justify-content: space-evenly;
        border: var(--border-styling);
        color: var(--text);
        cursor: pointer;
        border-radius: var(--border-radius-inner);
        &:active {
            background: var(--bg-btn-active);
        }
    }

    .active {
        background: var(--bg-selected);
        border: var(--border-styling-active);
        &:active {
            background: var(--bg-btn-active);
        }
    }
</style>