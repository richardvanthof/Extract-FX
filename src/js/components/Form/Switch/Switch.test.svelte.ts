import { describe, it, vi, expect } from "vitest";
import { fireEvent, render, screen } from '@testing-library/svelte';
import Switch from './Switch.svelte';

describe('Switch',() => {
    it('Renders', () => {
        const props = {
            value: 'file',
            options: ['File', 'Sequence'],
            callback: vi.fn()
        }
        
        const { container } = render(Switch, { props })

        expect(container).toMatchSnapshot()

    })
    it('Check if switch works', async () => {
        const props = {
            value: 'file',
            options: ['File', 'Sequence'],
            callback: vi.fn((selection: string) => {
                props.value = selection;
            })
        }
        
        const {container, rerender} = render(Switch, { props })

        const sequenceBtn = screen.getByText('Sequence');

        await fireEvent.click(sequenceBtn);
        
        expect(props.callback).toHaveBeenCalledTimes(1);
        expect(container).toMatchSnapshot() // File is active
        
        await rerender({ value: props.value })

        expect(props.callback).toHaveBeenCalledWith('sequence');
        expect(container).toMatchSnapshot() // Sequence is active
    });
});