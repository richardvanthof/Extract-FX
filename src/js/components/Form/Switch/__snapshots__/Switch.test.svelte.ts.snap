// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Switch > Check if switch works 1`] = `
<div>
  <ul
    class="switch-input svelte-2z4jz9"
  >
    <button
      class="switch-option active svelte-2z4jz9"
    >
      File
    </button>
    <button
      class="switch-option  svelte-2z4jz9"
    >
      Sequence
    </button>
    
  </ul>
  
</div>
`;

exports[`Switch > Check if switch works 2`] = `
<div>
  <ul
    class="switch-input svelte-2z4jz9"
  >
    <button
      class="switch-option  svelte-2z4jz9"
    >
      File
    </button>
    <button
      class="switch-option active svelte-2z4jz9"
    >
      Sequence
    </button>
    
  </ul>
  
</div>
`;

exports[`Switch > Renders 1`] = `
<div>
  <ul
    class="switch-input svelte-2z4jz9"
  >
    <button
      class="switch-option active svelte-2z4jz9"
    >
      File
    </button>
    <button
      class="switch-option  svelte-2z4jz9"
    >
      Sequence
    </button>
    
  </ul>
  
</div>
`;
